### DISCUSSION TITLE
Touchdisplay for the selection of audiobook

### DISCUSSION THREAD

**Discussion by PsychoD** [MiczFlor/RPi-Jukebox-RFID#1450] [Q&A]
Hi,

with the rising need of my children for more and more audiobooks on the phoniebox I notice, that I am the limiting factor for endless listening pleasure when it comes to designing, printing and laminating RFID cards. So currently I have to start most audiobooks by Webinterface, since many cards are not created yet. This annoys me and much more serious, annoys my children :-)

It would be great to install a touchscreen into the phoniebox, which shows "artists" after booting up, so children can select "artists" first and the episode in a second step. 

Has something like this already been done or is it possible?

Best regards
Psy

**Comment by Davepre** [comment]
I bought this touch screen for exactly the purpose you described but I haven't had the time to implement it yet. I was looking for a small touch screen so this seems like a good fit for me:

[https://de.aliexpress.com/item/1005001659605564.html?spm=a2g0s.9042311.0.0.31154c4dlO5weZ](url)

**Comment by PsychoD** [comment]
Sounds great! Could you please fix the link? It is not working currently. 

In the meantime I found the Sonos Kids controller project, which is pretty ideal regarding the usability I think: 
https://forum-raspberrypi.de/forum/thread/48811-sonos-kids-controller-nun-auch-fuer-andere-spotify-connect-geraete-touchscreen-j/

GitHub:
https://github.com/Thyraz/Sonos-Kids-Controller

Maybe it could be a starting point for us to reuse some of that code and build a „connector“ for the phoniebox?

**Reply by Davepre** [reply]
thats odd. I fixed the link but it only seems to work when I copy/paste it in the browser. I ordered it on Ali Express where its called "Waveshare 2.8inch Capacitive Touch Screen LCD for Raspberry Pi, 480*640, DPI, IPS, Fully Laminated Toughened Glass Cover, Low Po"

**Reply by pabera** [reply]
> Maybe it could be a starting point for us to reuse some of that code and build a „connector“ for the phoniebox?

I like this idea. Sonos would than need to act just like another sound output.

**Reply by PsychoD** [reply]
I am not sure if Sonos could be an output for the phoniebox, but if somehow mplayer (or whatever is currently used) can be configured to play on sonos, then yes. 

I was mainly focussed on the UI and how it could be used by a 4-year old. They think like: "Hey, I would like to listen to Paw Patrol". So first we'd need a list of artists or audio book series. The child selects an artist and then gets a list of all episodes, again visualized by their covers. After clicking on an episode, the episodes starts. Since most phonieboxes already bring start/stop controls as well as volume (which are easier to operate for children than touch), for the first step it would not even be crucial to have touch controls for that. It would be nice to just show the cover of the currently playing episode and maybe the position in the episode (although that again would be something a child would not be interested in). 

I did not have time to look at the phoniebox UI code yet, but I guess we would need to see what kind of interfaces there are to get a) a list of artists, b) get a list of episodes per artist and c) analyze how the sonos kids controller currently generates the UI to find a way to adapt. 

By the way, I think it could work to skip thinking about artists and episodes, any just think in folders. So the UI would only need to navigate through folders (which contain the known cover.jpg) and when there are .mp3s in the folder, they are played.

**Comment by PsychoD** [comment]
Hi, 

my son turned 3, and so it was time for the next Phoniebox build - Phoniebox touch. :-) I have integrated a waveshare 2,8" capacitive touch display, which works really, really well. I used the Future3 branch, which lacks quite a few features, but the UI works out of the box with the touch display. I think it needs some optimizations for "kid-kiosk-mode", but I have never worked with react, so that will take some time. 

Here a little teaser: 
![IMG_5794](https://user-images.githubusercontent.com/15204202/151149440-49c1c192-027d-466d-aa50-3506fbf0c761.jpg)
![IMG_5795](https://user-images.githubusercontent.com/15204202/151149447-0b244dd8-3a30-4b5c-9b53-ae7567f44a38.jpg)
![IMG_5791](https://user-images.githubusercontent.com/15204202/151149455-07b31c1d-cd18-46e6-a528-047cbb70b866.jpg)

**Reply by pabera** [reply]
I'm truly pleased to see your cool example. When building the new Webapp, the Kioskmode was a low hanging fruit to put in. But you are right, the UX is optimized for mobile usage, or more concrete, it's dedicated to an adult audience.

When reading your post and looking at the photos, it became immediately obvious that the UI, at least in the kiosk mode, needs to be simpler and much easier to use for the kids.

So thanks for sharing.. and stay tuned for updates. 😀

**Reply by pabera** [reply]
Just another question.. you mentioned the future3 branch lacks features. That's probably true.. I am interested to what these features are that you are missing?

**Reply by ghost** [reply]
Execellent work! Great to see Version 3 in operation. 

Could you share some details of the touch display and a link where to obtain it?

**Reply by PsychoD** [reply]
> 

First of all: Thanks for the collaboration and all your efforts. This really is a great project and the phoniebox is in heady use every single day. 

I started with the kiosk mode, but did not get it working properly with the touch display. The display's view and touch input needs to be rotated (it is portrait by default) and I didn't manage to. I did only manage to do so in bullseye desktop. So I based the install on the desktop-version of bullseye and autostart chromium pointing to localhost. Running of a RPI Zero2 it is quite slow unfortunately, but besides that it works well. 

Regarding the UI: I am so excited for further updates, do you already have an idea? If I can help, please let me know. :-) I already had a brief look at the code and had already sketched out my idea before my limited react knowledge crossed my way. If you are interested, I could share it. The good thing is, children need much less text, much less features but more of big pictures and covers. A possibility to read out a folder name with text to speech would be great to memorize the name of an album for each cover and a hidden button could switch to the current parent view.

**Reply by PsychoD** [reply]
> Just another question.. you mentioned the future3 branch lacks features. That's probably true.. I am interested to what these features are that you are missing?

What I meant in the first place was, that not all known functions from V2 are already implemented in the current main branch and it might still be unstable. No. 1 on my feature wishlist would really be the kid-compatible touch display support, including a text to speech-button to read the title of a folder/album. What I did in my V1 Phoniebox build was to control LEDs of buttons etc. using PWM in the python scripts. This way I could activate a night mode, that dimmed all buttons or even switched them off. How could I do that in V3? Maybe that would be a good feature?

**Reply by PsychoD** [reply]
> Execellent work! Great to see Version 3 in operation.
> 
> Could you share some details of the touch display and a link where to obtain it?

Thanks so much! 
Interestingly, I chose the same display that Davepre mentioned above - but by pure coincidence: 
https://www.waveshare.com/2.8inch-hdmi-lcd-h.htm

I ordered it off amazon since I needed it quickly: 
https://www.amazon.de/gp/product/B09ND89R16/ref=ppx_yo_dt_b_asin_title_o08_s00?ie=UTF8&th=1

I am very satisfied with it. Capacitive touch works well even with parent fingers and it is very nice to look at. O course it is a bit small but I did not have more space in my built. There are larger versions, but my second goal was to minimize power drain. 

I will share details regarding the build in a separate post, maybe it helps someone.

**Reply by ghost** [reply]
> What I did in my V1 Phoniebox build was to control LEDs of buttons etc. using PWM in the python scripts. This way I could activate a night mode, that dimmed all buttons or even switched them off. How could I do that in V3? Maybe that would be a good feature?

Interesting feature! We are currently shifting the entire GPIO handling towards a new implementation based on GPIOZero. With that it will be relatively easy to implement this feature. The next release will have GPIOZero - but not the night time feature yet. But it's probably due soon now...

**Reply by PsychoD** [reply]
> > What I did in my V1 Phoniebox build was to control LEDs of buttons etc. using PWM in the python scripts. This way I could activate a night mode, that dimmed all buttons or even switched them off. How could I do that in V3? Maybe that would be a good feature?
> 
> Interesting feature! We are currently shifting the entire GPIO handling towards a new implementation based on GPIOZero. With that it will be relatively easy to implement this feature. The next release will have GPIOZero - but not the night time feature yet. But it's probably due soon now...

Sounds good! I am waiting eagerly for any new release :-))

**Reply by PsychoD** [reply]
Hello everybody, I just wanted to ask if there are any news regarding Touch-UI, of if anything in that regard is planned.

**Comment by PsychoD** [comment]
One more thing I noticed: The UI on the touch screen is very, very slow. It runs only on a RPI Zero 2 (so basically a RPI3) of course, but I also noticed that Chromium is eating up much CPU time: 
![grafik](https://user-images.githubusercontent.com/15204202/151701386-91dd14cf-98b2-4b54-bd05-bfe0379f30b3.png)

On the box, the "Start" Tab is active, and audio is playing.

**Comment by hailogugo** [comment]
Which file do you open on the screen. I had the kiok mode run with the index lcd.php but that is only a screen to controll the player.

My Question which file do you use to open it in KIOSK Mode?

Now I understand. You install the future 3 fork with Kiosk Mode. 
I testing it but I do not want Settings on the touchscreen. I will look up if I can change it.

**Reply by PsychoD** [reply]
Sorry I missed your question. I am starting chromium in kiosk mode like this: 

sudo nano /etc/xdg/lxsession/LXDE-pi/autostart

@chromium-browser --incognito --disable-pinch --overscroll-history-navigation=0 --noerrdialogs --disable-translate --no-first-run --fast --fast-start --disable-infobars --kiosk http://localhost

**Comment by hm010120** [comment]
Hello,
I really like the Phoniebox Touch project and its implementation. It perfectly meets my children's current usability needs. Since I’m not the most experienced user when it comes to Raspberry Pi and similar devices, I wanted to ask if there is a tutorial or a GitHub repository available for integrating the touch display and kiosk mode?
Regards,
Marco

**Reply by PsychoD** [reply]
Due to the problems I had, I did not follow this further until now :(

**Reply by hm010120** [reply]
Thanks for your response :-) I will try by myself

**Reply by bittu** [reply]
@hm010120 I'm also in the same situation of finding a display and tutorial to set it up for track selection. Did you come across any, please do let me know.
Thanks

**Reply by PsychoD** [reply]
I hope this is  "allowed", but there is a different project called MuPiBox, that does exactly that: https://mupibox.de/#

**Reply by hm010120** [reply]
> @hm010120 I'm also in the same situation of finding a display and tutorial to set it up for track selection. Did you come across any, please do let me know. Thanks

Not yet, but I have invested a little time in a control unit for a display. Unfortunately, I still don't had enough time to test the hardware, but Easter is just around the corner.

**Reply by hm010120** [reply]
> I hope this is "allowed", but there is a different project called MuPiBox, that does exactly that: https://mupibox.de/#

Thank you for the hint. I had already discovered the project and tried it out. I would prefer to upgrade the Phoniebox here (I currently have 4 Phonieboxes in use and I would like to upgrade 2 of them) so that the system administration and synchronization is even clearer for my family.
